mountpoint-s3/src/main.rs

```diff
@@ -340,6 +340,13 @@ fn mount(args: CliArgs) -> anyhow::Result<FuseSession> {
         .transpose()
         .context("Failed to parse endpoint URL")?;
 
+    let ec2_instance_type = retrieve_instance_type().unwrap_or_else(|_| {
+        tracing::error!("Unable to detect instance type.");
+        String::from("unknown instance type.")
+    });
+
+    tracing::debug!("the ec2 instance type is: {}", &ec2_instance_type);
```

Comment:
Seems like we'd prefer `ec2_instance_type` to just be `Option<String>` here to represent the failure case.

Reply:
Changed